internal/controller/ols_app_postgres_assets.go

```diff
@@ -454,7 +454,11 @@ func (r *OLSConfigReconciler) storageDefaults(s *olsv1alpha1.Storage) error {
 
 func (r *OLSConfigReconciler) generatePostgresPVC(cr *olsv1alpha1.OLSConfig) (*corev1.PersistentVolumeClaim, error) {
 
-	storage := cr.Spec.OLSConfig.Storage
+	// Create a copy of the storage configuration to avoid modifying the original CR
+	storage := &olsv1alpha1.Storage{}
+	if cr.Spec.OLSConfig.Storage != nil {
+		storage = cr.Spec.OLSConfig.Storage.DeepCopy()
+	}
```

Comment:
@sriroopar Not sure what triggered to make the above lines part of this PR - looks unrelated.